### ISSUE TITLE
i18n migration collation. Make it utf8_bin instead of utf8_unicode_ci

### ISSUE THREAD

**Issue by ruskid** [yiisoft/yii2#15280]
It's more of a suggestion/question.

When you create tables **source_mesage** and **message** with Yii migration the default collation used is **utf8_unicode_ci**

1. MySQL treats strings as accent insensitive. "Revolución" is same as "Revolucion".
2. Now in Yii2 isset in array is used to find translations.
```php 
if (isset($this->_messages[$key][$message]) && $this->_messages[$key][$message] !== '') {
            return $this->_messages[$key][$message];
        }
```
Inconsistency, what happened to me is that in my back office there is a translation for "Revolución", cause MySQL returned results for it. But in front office, Yii fails to source language cause of php isset.

My question/proposition is to change the collation to utf8_bin on **source_message.message**. So in the back office, there won't be any misleading translation loaded.

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.12?
| PHP version      | 
| Operating system |

**Comment by Renkas** [comment]
That's just a default. You can use whatever collation you want for your app.

I can't really see the need to change this on framework level.

**Comment by ruskid** [comment]
ofc I can use my collation and I already use.

I disagree. 
1. By changing migration you don't change the framework. 
2. It makes more sense and a time saver for others.

The default doesn't mean best. I personally lost some time wondering why some translations were not loaded.

**Comment by Renkas** [comment]
Using utf8_bin should be done ONLY when you the developer knowingly decide so. It has many disadvantages and is not suitable as a default value.

Do you even know all the disadavantages/differences?

**Comment by Renkas** [comment]
Well - I read now that you mean to change only the migration of i18n. Thet's different thing. At first I understood you want to change the migration template.

In this case your proposal is not all bad and utf8_bin seems to suit there well.

Sorry for the misunderstanding.

**Comment by ruskid** [comment]
Right, no problem. The issue is that **php isset in array** is case sensitive, when getting the translation. 
So there is no actual use of utf8_unicode_ci.
